Jackson version conflict with avatica

There is a conflict with jackson-databind 2.6.3 (from transitive dependencies of calcite-core and avatica-core) and druid's explicit jackson-databind 2.4.6 dependency. I see issues during batch ingestion. Realtime using tranquility is fine. Here is a gist containing the stack trace

Druid has direct dependency on `calcite-core`, `avatica-core` and `avatica-server` from https://github.com/druid-io/druid/pull/3682.

`avatica-core` is the non-shaded artifact which only contains client classes without any bundled dependencies except jackson. as per this: https://github.com/apache/calcite/commit/e5a42f1ca80eca0bb6f0c8f6677f106fadb2e652#diff-eb4d33b91f63eeaaea2ffd36688b61f8R49

`calcite-core` depends on shaded avatica.
`avatica-server` depends on non-shaded avatica
`avatica-core` is the non-shaded avatica.

cc: @gianm

I'm not able to build tip, is this the same issue as #3800 or different?  seems that `calcite-core` is mentioned....

I'm osx 10.11.6 with java 1.8.0_66

~~~
...
[INFO] druid-time-min-max ................................. SKIPPED
[INFO] druid-google-extensions ............................ SKIPPED
[INFO] druid-virtual-columns .............................. SKIPPED
[INFO] druid-thrift-extensions ............................ SKIPPED
[INFO] ambari-metrics-emitter ............................. SKIPPED
[INFO] druid-caffeine-cache ............................... SKIPPED
[INFO] distribution ....................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:17 min
[INFO] Finished at: 2016-12-23T19:21:36-06:00
[INFO] Final Memory: 31M/437M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project druid-sql: Could not resolve dependencies for project io.druid:druid-sql:jar:0.9.3-SNAPSHOT: Failed to collect dependencies at org.apache.calcite:calcite-core:jar:1.10.0 -> org.pentaho:pentaho-aggdesigner-algorithm:jar:5.1.5-jhyde: Failed to read artifact descriptor for org.pentaho:pentaho-aggdesigner-algorithm:jar:5.1.5-jhyde: Could not transfer artifact org.pentaho:pentaho-aggdesigner-algorithm:pom:5.1.5-jhyde from/to conjars (http://conjars.org/repo): Connect to conjars.org:80 [conjars.org/107.22.169.25] failed: Operation timed out -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
j ~/go/src/github.com/druid-io/druid (master) $ 
~~~

--- a/pom.xml
+++ b/pom.xml
@@ -59,8 +59,8 @@
 
     <properties>
         <apache.curator.version>2.11.1</apache.curator.version>
-        <avatica.version>1.8.0</avatica.version>
-        <calcite.version>1.10.0</calcite.version>
+        <avatica.version>1.9.0</avatica.version>
+        <calcite.version>1.11.0</calcite.version>
         <guava.version>16.0.1</guava.version>
         <guice.version>4.1.0</guice.version>
         <jetty.version>9.2.5.v20141112</jetty.version>
@@ -261,6 +261,11 @@
                 <groupId>org.apache.calcite</groupId>
                 <artifactId>calcite-linq4j</artifactId>
                 <version>${calcite.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.apache.calcite.avatica</groupId>
+                <artifactId>avatica-core</artifactId>
+                <version>${avatica.version}</version>
             </dependency>
             <dependency>
                 <groupId>org.apache.calcite.avatica</groupId>

--- a/sql/pom.xml
+++ b/sql/pom.xml
@@ -48,6 +48,10 @@
     </dependency>
     <dependency>
       <groupId>org.apache.calcite.avatica</groupId>
+      <artifactId>avatica-core</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.calcite.avatica</groupId>
       <artifactId>avatica-server</artifactId>
     </dependency>
     <dependency>

--- a/sql/src/main/java/io/druid/sql/calcite/planner/DruidPlannerImpl.java
+++ b/sql/src/main/java/io/druid/sql/calcite/planner/DruidPlannerImpl.java
@@ -27,6 +27,7 @@ import org.apache.calcite.plan.RelOptRule;
 import org.apache.calcite.plan.volcano.VolcanoPlanner;
 import org.apache.calcite.prepare.CalcitePrepareImpl;
 import org.apache.calcite.rel.RelCollationTraitDef;
+import org.apache.calcite.sql2rel.SqlRexConvertletTable;
 
 /**
  * Our very own subclass of CalcitePrepareImpl, used to alter behaviors of the JDBC driver as necessary.
@@ -64,5 +65,11 @@ public class DruidPlannerImpl extends CalcitePrepareImpl
     }
 
     return planner;
+  }
+
+  @Override
+  protected SqlRexConvertletTable createConvertletTable()
+  {
+    return DruidConvertletTable.instance();
   }
 }

--- a/sql/src/test/java/io/druid/sql/calcite/CalciteQueryTest.java
+++ b/sql/src/test/java/io/druid/sql/calcite/CalciteQueryTest.java
@@ -106,9 +106,6 @@ public class CalciteQueryTest
 {
   private static final Logger log = new Logger(CalciteQueryTest.class);
 
-  // Used to mark tests that should pass once Calcite 1.11.0 is released.
-  private static final boolean CALCITE_1_11_0 = false;
-
   private static final PlannerConfig PLANNER_CONFIG_DEFAULT = new PlannerConfig();
   private static final PlannerConfig PLANNER_CONFIG_NO_TOPN = new PlannerConfig()
   {
@@ -588,11 +585,6 @@ public class CalciteQueryTest
   @Test
   public void testGroupByNothingWithLiterallyFalseFilter() throws Exception
   {
-    if (!CALCITE_1_11_0) {
-      // https://issues.apache.org/jira/browse/CALCITE-1488
-      return;
-    }
-
     testQuery(
         "SELECT COUNT(*), MAX(cnt) FROM druid.foo WHERE 1 = 0",
         ImmutableList.<Query>of(),
@@ -605,11 +597,6 @@ public class CalciteQueryTest
   @Test
   public void testGroupByOneColumnWithLiterallyFalseFilter() throws Exception
   {
-    if (!CALCITE_1_11_0) {
-      // https://issues.apache.org/jira/browse/CALCITE-1488
-      return;
-    }
-
     testQuery(
         "SELECT COUNT(*), MAX(cnt) FROM druid.foo WHERE 1 = 0 GROUP BY dim1",
         ImmutableList.<Query>of(),
@@ -2002,11 +1989,6 @@ public class CalciteQueryTest
   @Test
   public void testGroupByExtractYear() throws Exception
   {
-    if (!CALCITE_1_11_0) {
-      // https://issues.apache.org/jira/browse/CALCITE-1509
-      return;
-    }
-
     testQuery(
         "SELECT\n"
         + "  EXTRACT(YEAR FROM __time) AS \"year\",\n"
@@ -2053,11 +2035,6 @@ public class CalciteQueryTest
   @Test
   public void testExtractFloorTime() throws Exception
   {
-    if (!CALCITE_1_11_0) {
-      // https://issues.apache.org/jira/browse/CALCITE-1509
-      return;
-    }
-
     testQuery(
         "SELECT\n"
         + "EXTRACT(YEAR FROM FLOOR(__time TO YEAR)) AS \"year\", SUM(cnt)\n"
@@ -2272,11 +2249,6 @@ public class CalciteQueryTest
   @Test
   public void testUsingSubqueryAsFilterOnTwoColumns() throws Exception
   {
-    if (!CALCITE_1_11_0) {
-      // https://issues.apache.org/jira/browse/CALCITE-1479
-      return;
-    }
-
     testQuery(
         "SELECT __time, cnt, dim1, dim2 FROM druid.foo "
         + " WHERE (dim1, dim2) IN ("

